scripts/inject-metadata

```diff
@@ -50,22 +51,26 @@ if __name__ == "__main__":
             if namespace_defaults is False:
                 # A namespace defaults file that sets '"mozPipelineMetadata" = false'
                 # is a special flag to exempt the namespace from metadata injection.
-                inject_metadata = False
+                inject_default_metadata = False
             else:
+                propagate_metadata = True
                 merge_where_empty(meta, namespace_defaults)
+    else:
+        propagate_metadata = True
```

Comment:
This accurately preserves the existing behavior of keeping or creating an empty `mozPipelineMetadata` dict if the namespace has no `defaults.schema.json` file, but is that actually desired behavior?

If not, this could be removed, letting the preceding `propagate_metadata = bool(meta)` determine whether `mozPipelineMetadata` is included in the output.